### PR TITLE
Add helm-converter tool for converting kube-compare reference configs  to Helm charts

### PR DESCRIPTION
**Pull request by AlinaSecret**
- Implement helm-converter tool to convert kube-compare  reference configs into valid Helm charts
- Include documentation for using the helm-convertor tool to create and render Helm charts
- Provided easy creation of `values.yaml` by gathering  automatically from templates  list of required variables for template rendering  
- Provide feature for auto-extracting default values from existing CRs

for complete context and docs view README in PR

TODO:

1. tests
2. tidy up (code + command line help + func docs)

### CODE REVIEW

File: addon-tools/compare-to-helm/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+The initial implementation of the values gathering is a great place to start with, and for complex cases users may will
+need to so some minor modifications. This i also a one time thing per reference, for references that have already been converted
+the previous values.yaml can be used as a base for the values.yaml file. Most of the work is in the initial conversion.
+
+example generated values.yaml structure (does not include the default values):
+
+```yaml
+#gnerated from: pkg/compare/testdata/AllRequiredTemplatesExistAndThereAreNoDiffs/reference/metadata.yaml
+cm:
+- {} # Template with no user defined content
```
Comment (nocturnalastro) [inline addon-tools/compare-to-helm/README.md:74]: Is a template has no used defined data why not just skip it.

**Review by lack — COMMENTED** [review]
Architecturally, would it make sense for this to be part of the main executable as a sub-command?  

Rather than `helm-convert x y z` do we want `kubectl cluster-compare to-helm x y z`?

File: addon-tools/compare-to-helm/README.md (hunk 1 above)
Comment (AlinaSecret) [reply addon-tools/compare-to-helm/README.md]: yeah, your right changed it

File: addon-tools/compare-to-helm/README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+# create helm chart
+helm-convert -r ./reference/metadata.yaml -n ChartDir 
+# Render templates, resulting in creation of default CRs that follow the reference 
+helm template renderedRef ChartDir/ChartDir/ --output-dir renderedRefDir
+```
+
+Auto Extracting of default values from Existing CRs:
+
+```shell
+helm-convert -r ./reference/metadata.yaml -n ChartDir -d ExstisingCRsDir
```
[the remark this answers is not in the document]
Comment (AlinaSecret) [reply addon-tools/compare-to-helm/README.md]: nothing, at least as i implemented it because it may contain CRs that are not supposed to be in the reference or different unrelated yaml files .
Is there any behavior you would of expect?

**Comment by AlinaSecret** [comment]
> Architecturally, would it make sense for this to be part of the main executable as a sub-command?
> 
> Rather than `helm-convert x y z` do we want `kubectl cluster-compare reference-to-helm x y z`?

i think this can be a good direction, like you said architecturally. not sure about what it means product wise, like do we need to support this, and is it the best for the project. I remember discussing it with Ian when i added the report creator and dont  remember the conclusions. I think also it was something about still be able to add it to oc in the future.
any way if we do combine it, i think the same should be done for the  report creator. 
maybe its a good idea to rediscuss this ( Micky also asked the same about the report creator once)

File: addon-tools/compare-to-helm/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nocturnalastro) [reply addon-tools/compare-to-helm/README.md]: I would expect it to be added to as helm chart. That way you can keep it as part of the cycle.

**Comment by imiller0** [comment]
Nice work on this tool an PR. I'll add some additional thoughts/questions later but wanted to share some initial feedback first.

The tool does a really nice job of getting from the "annotated reference" (kube-compare compatible crs) to usable configuration CRs. The ability to quickly generate a baseline values.yaml, and to be able to consume defaults for those values from an existing config, is a huge enabler for using this tool. Really nice.

On the question of stand-alone or part of kube-compare. My initial thoughts are that merging the two would make it harder to clearly articulate what the tool does. Delivering them as separate tools means there is a clearly defined function that each performs (context aware diff versus conversion). Each can excel and grow as needed. I also think that the likelihood of bugs/issues in the helm-convert are more likely (interpreting the reference to generate values.yaml will be hard in the general case). Keeping the tools separate helps to identify which bugs are affecting which use case.

File: addon-tools/compare-to-helm/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (imiller0) [reply addon-tools/compare-to-helm/README.md]: I think there are a couple of interesting use cases here. We should articulate a set of user stories to cover this functionality to help guide how it is implemented. eg
- Pulling defaults from a directory to populate the values.yaml for an existing reference. May want to highlight crs with no existing reference for consideration to pull them in as new reference CRs.
- Pulling defaults from a live cluster
- etc.

**Review by lack — APPROVED** [review]
I think the only thing I'd like to see is more unit testing and more negative test cases.  The tests here are more system-level and positive cases, but there are gaps and code paths we're probably not exercising.

But that's probably not reason enough to block this code going in.
